packages/react-ui-components/src/components/SchemaForm/Widgets/CheckboxesWidget/utils.ts

```diff
@@ -77,6 +77,10 @@ export const toggleSelectedOptions = <T extends CheckboxOption>(
 	const allButtonsSelected = selectedOptions.length === allOptions.length;
 
 	if (allButton) {
+		if (selectedOptionValue !== ALL_BUTTON_VALUE && allButtonsSelected) {
```

Comment:
We can move this to below the `(selectedOptionValue === ALL_BUTTON_VALUE)` condition just like this:
```typescript
if (allButtonsSelected) {
    return [selectedOptionValue];
}
```